cidc_api/services/ingestion.py

```diff
@@ -440,3 +451,44 @@ def signed_upload_urls():
         object_urls[object_name] = object_url
 
     return jsonify(object_urls)
+
+
+@ingestion_api.route("/extra-assay-metadata", methods=["POST"])
+def extra_assay_metadata():
+    """
+
+    Extracts:
+        job_id, and extra_metadata_file from request body
+    Raises:
+        BadRequest: if the request requirements aren't satisfied
+
+    request.form = {
+        'job_id': the job_id to update the patch for,
+    }
+
+    request.files = {
+        [artifact_uuid_1]: [open extra metadata file 1],
+        [artifact_uuid_2]: [open extra metadata file 2]
+    }
+
+    """
+
+    if not request.form:
+        raise BadRequest("Expected form content in request body, or failed to parse form content")
+
+    if 'job_id' not in request.form:
+        raise BadRequest("Expected job_id in form")
+
+    if not request.files:
+        raise BadRequest("Expected files in request (mapping from artifact uuids to open files)")
+
+    job_id = request.form['job_id']
+
+    for uuid, file in request.files.to_dict().items():
+        try:
+            AssayUploads.merge_extra_metadata(job_id, uuid, file)
```

Comment:
I think would be better to merge them all in one, avoiding committing session multiple times.